### ISSUE TITLE
Error message - not taking the user back out the page

### ISSUE THREAD

**Issue by magdalenadrafiova** [DMPRoadmap/roadmap#2176]
From our accessibility testing it has been reported that we need to improve error messages on the page where the error occurs and not take the user back out the page. Also the error message should have improved the colour contrast which relates to #1457 and we should add format restrictions to fields - e.g. telephone numbers.

**Comment by johnpinto1** [comment]
Going to park this as I am not sure about where this is required.

**Comment by mariapraetzellis** [comment]
This is a major change and would require rethinking our error  process. At least a weeks worth of work. Might make sense to handle when we move to rails 5. @xsrust can add more details here.
